rollout/bluegreen.go

```diff
@@ -35,6 +35,12 @@ func (c *RolloutController) rolloutBlueGreen(r *v1alpha1.Rollout, rsList []*apps
 
 	roCtx := newBlueGreenCtx(r, newRS, oldRSs, arList)
 	logCtx := roCtx.Log()
+
+	err = c.podRestarter.Reconcile(roCtx)
```

Comment:
With this here, it seems like we still perform restarts over pod template changes. Isn't it desired that if we ever change the pod spec, we should cancel the restart request (since restarting pods is useless if we are about to move to a new spec)?

Reply:
I moved the reconcile calls to after the pod template check and made sure to set the restartedat time if the pod template has changed.